### PR TITLE
Throw a `TypeError` instead of the generic `Error`

### PR DESCRIPTION
**Pull request by rohit2sharma95**
Ref: https://github.com/twbs/bootstrap/pull/32376#discussion_r546599766

### CODE REVIEW

**Review by XhmikosR — DISMISSED** [review]

**Comment by XhmikosR** [comment]
@rohit2sharma95 actually, shouldn't this test fail right now?

https://github.com/twbs/bootstrap/blob/206b9f45f0b91fef62ac18470165d91748ade161/js/tests/unit/util/index.spec.js#L213-L215

Perhaps we should use `toThrowError`?

File: js/src/dropdown.js [hunk 1]
```diff
@@ -263,7 +263,7 @@ class Dropdown extends BaseComponent {
       typeof config.reference.getBoundingClientRect !== 'function'
     ) {
       // Popper virtual elements require a getBoundingClientRect method
-      throw new Error(`${NAME}: Option "reference" provided type "object" without a required "getBoundingClientRect" method.`)
+      throw new TypeError(`${NAME}: Option "reference" provided type "object" without a required "getBoundingClientRect" method.`)
```
Comment (XhmikosR) [inline js/src/dropdown.js:266]: I wonder if we should make the related tests stricter, probably makes sense.

Comment (rohit2sharma95) [reply js/src/dropdown.js]: Updated.
Also, changed the `NAME` to upper case to make the consistency.
Should check for the same in `jQueryInterface` also? :thinking:

Comment (XhmikosR) [reply js/src/dropdown.js]: What do you mean?

Comment (rohit2sharma95) [reply js/src/dropdown.js]: I mean should the tests be updated here? (To check if the `error` is the instance of `TypeError`)
https://github.com/twbs/bootstrap/blob/7e6d7b8f817ad1212238f15124335d37c93ebfea/js/tests/unit/dropdown.spec.js#L1560-L1564

Comment (XhmikosR) [reply js/src/dropdown.js]: Ah, yeah let's switch any other instances to `toThrowError`

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**